editor/import/3d/scene_import_settings.cpp

```diff
@@ -233,6 +250,21 @@ void SceneImportSettingsDialog::_fill_material(Tree *p_tree, const Ref<Material>
 	MaterialData &material_data = material_map[import_id];
 	ERR_FAIL_COND(p_material != material_data.material);
 
+	Variant value;
+	if (_get_current("materials/extract", value) && (int)value != 0) {
+		String spath = base_path.get_base_dir();
+		if (_get_current("materials/extract_path", value)) {
+			String extpath = value;
+			if (!extpath.is_empty()) {
+				spath = extpath;
+			}
+		}
+		String ext = (_get_current("materials/extract_format", value) && (int)value == 1) ? ".res" : ".tres";
+		String path = spath.path_join(import_id.validate_filename() + ext);
+		material_data.settings["use_external/enabled"] = true;
+		material_data.settings["use_external/path"] = path;
```

Comment:
Same problem: should write fallback_path in caae the uid was lost and we need to lookup by res:// . Another case of duplicate logic which implies more maintenance burden